deploy/openshift/templates/che-monitoring.yaml

```diff
@@ -4193,13 +4196,99 @@ objects:
               "alignLevel": null
             }
           },
+          {
+            "aliasColors": {},
+            "bars": false,
+            "dashLength": 10,
+            "dashes": false,
+            "datasource": "Che",
+            "description": "Percent of successfully started workspaces with 1 hour period",
+            "fill": 1,
+            "gridPos": {
+              "h": 5,
+              "w": 11,
+              "x": 0,
+              "y": 6
+            },
+            "id": 36,
+            "legend": {
+              "avg": true,
+              "current": true,
+              "max": true,
+              "min": true,
+              "show": true,
+              "total": false,
+              "values": true
+            },
+            "lines": true,
+            "linewidth": 1,
+            "links": [],
+            "nullPointMode": "null",
+            "paceLength": 10,
+            "percentage": false,
+            "pointradius": 2,
+            "points": false,
+            "renderer": "flot",
+            "seriesOverrides": [],
+            "stack": false,
+            "steppedLine": false,
+            "targets": [
+              {
+                "expr": "100*(1 - (sum(increase(che_workspace_failure_total{while=\"STARTING\"} [1h]))/sum(increase(che_workspace_started_total [1h]))))",
```

Comment:
Is this the correct expression for the success rate? 

What will be the result if I start with 0 workspaces and attempt to start 5 of them and all of the starts fail? I expect the below expression would yield `NaN` whereas I'd expect 0%...

```
100*(1 - (sum(increase(che_workspace_failure_total{while=\"STARTING\"} [1h]))/sum(increase(che_workspace_started_total [1h]))))
```

What about changing it to (pulled out of thin air with no syntax check, so please take it with a grain of salt):
```
100 * sum(increase(che_workspace_started_total [1h]))/sum(increase(che_workspace_starting_total [1h]))
```

Reply:
Actually I must be missing something because in your example with 1 failed workspace and 2 running, how can your expression give 75%?

Reply:
>Actually I must be missing something because in your example with 1 failed workspace and 2 running, how can your expression give 75%?

AFAIK. Numbers are aggregated in the 1h period. I assume I have 1 failed and 4 started workspaces during the last hour